ibis/expr/tests/test_datatypes.py

```diff
@@ -335,3 +335,34 @@ def test_time():
 
 def test_time_valid():
     assert dt.validate_type('time').equals(dt.time)
+
+
+# implicit casting
```

Comment:
The test below is explicitly casting. Did you mean something else here?

Reply:
right, but first these DO can can_implicit_cast. better way of testing?

Reply:
This comment doesn't match what the test does. The test is checking that explicit casts don't raise an exception.

Implicit casts are conceptually similar to implicit casts in C. For example, saying `uint64_t x = 1` will automatically promote `1` to `uint64_t` without needed to explicitly say "do the cast".

If you want to test implicit casting then call the `can_implicit_cast` method on `DataType` instances, e.g., `dt.int8.can_implicit_cast(dt.int16)`.

Reply:
Furthermore, explicit casts between scalar numeric casts should always be valid ibis expressions, whereas there are many types that are not implicitly castable to many other types, e.g., `float32` is not implicitly castable to `int8`.

Reply:
I never said this was a great test. We are *not* testing casting really anywhere in the codebase. This should certainly be a followup. This is just a smoketest at this time.

Reply:
My only issue here is that the comment needs to be changed or removed since it doesn't match what the test is doing.

> We are not testing casting really anywhere in the codebase.

We have around 40 tests for *explicit* casting but probably 0 or just a few for *implicit* casting. They are meaningfully different, so we shouldn't gloss over it in the code.

Reply:
removed the tests entirely.